pkg/metrics/bundle_metrics.go

```diff
@@ -0,0 +1,123 @@
+)
+
+var (
+	bundleSubsystem = "bundle"
+	bundleLabels    = []string{"name", "namespace"}
+
+	bundleNotReadyDeployments = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: bundleSubsystem,
+			Name:      "not_ready",
+			Help:      "Number of deployments for a specific bundle in a not ready state.",
+		},
+		bundleLabels,
+	)
+	bundleWaitAppliedDeployments = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: bundleSubsystem,
+			Name:      "wait_applied",
+			Help:      "Number of deployments for a specific bundle in a wait applied state.",
+		},
+		bundleLabels,
+	)
+	bundleErrAppliedDeployments = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: bundleSubsystem,
+			Name:      "err_applied",
+			Help:      "Number of deployments for a specific bundle in a error applied state.",
+		},
+		bundleLabels,
+	)
+	bundleOutOfSyncDeployments = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: bundleSubsystem,
+			Name:      "out_of_sync",
+			Help:      "Number of deployments for a specific bundle in a out of sync state.",
+		},
+		bundleLabels,
+	)
+	bundleModifiedDeployments = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: bundleSubsystem,
+			Name:      "modified",
+			Help:      "Number of deployments for a specific bundle in a modified state.",
+		},
+		bundleLabels,
+	)
+	bundleReadyDeployments = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: bundleSubsystem,
+			Name:      "ready",
+			Help:      "Number of deployments for a specific bundle in a ready state.",
+		},
+		bundleLabels,
+	)
+	bundlePendingDeployments = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: bundleSubsystem,
+			Name:      "pending",
+			Help:      "Number of deployments for a specific bundle in a pending state.",
+		},
+		bundleLabels,
+	)
+	bundleDesiredReadyDeployments = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: bundleSubsystem,
+			Name:      "desired_ready",
+			Help:      "Number of deployments that are desired to be ready for a bundle.",
+		},
+		bundleLabels,
+	)
+	bundleObserved = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: bundleSubsystem,
+			Name:      "total_observations",
+			Help:      "The total times that this bundle has been observed",
+		},
+		bundleLabels,
+	)
+	bundleState = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: bundleSubsystem,
+			Name:      "state",
+			Help:      "Shows the state of this bundle deployment. Ready = 1, NotReady = 2, Pending = 3, OutOfSync = 4, Modified = 5, WaitApplied = 6, ErrApplied = 7.",
+		},
+		bundleLabels,
+	)
+)
+
+func ObserveBundle(bundle *fleet.Bundle, status *fleet.BundleStatus) {
```

Comment:
The metrics collection functions within the exporters are typically name (or prefixed) as `Collect()`. See https://prometheus.io/docs/instrumenting/writing_exporters/#collectors.

Reply:
Updated name, is now `CollectBundleMetrics` and the other methods follow the same pattern.